feature-libs/cart/main/root/components/mini-cart/mini-cart-component.service.ts

```diff
@@ -84,11 +82,11 @@ export class MiniCartComponentService {
    * Once the observable returned by activeCartRequired emits true, it completes.
    * activeCartRequired helps to make the mini cart compatible with some level of lazy loading.
    */
-  protected activeCartRequired() {
+  protected activeCartRequired(): Observable<boolean> {
```

Comment:
Each time, this function is called, there is a new stream created (which make `isFacadeImplProvided` is called). Can we just create one stream for it and maybe use `shareReplay`? something like this:
`protected activeCartRequired: Observable<boolean> = combineLatest...`?

Reply:
Good suggestion. I lean however to still leave it as-is for now, but lets keep it in mind for a future improvement.  There is no resource intensive processing being done by the stream.  Moreover, the page header is not recreated each time a user navigates to a new page in the storefront.  Since the mini cart is typically in the the header, this lowers the amount of calls made.